packages/babel-types/src/definitions/typescript.ts

```diff
@@ -91,13 +91,20 @@ defineType("TSQualifiedName", {
 
 const signatureDeclarationCommon = {
   typeParameters: validateOptionalType("TSTypeParameterDeclaration"),
-  parameters: validateArrayOfType(["Identifier", "RestElement"]),
-  typeAnnotation: validateOptionalType("TSTypeAnnotation"),
+  [process.env.BABEL_8_BREAKING ? "params" : "parameters"]: validateArrayOfType(
```

Comment:
Q: don't we use BABEL_TYPES_8_BREAKING in babel-types?

Reply:
Oh I didn't know `BABEL_TYPES_8_BREAKING`  environment variables.

Reply:
I prefer `BABEL_8_BREAKING` here. `BABEL_TYPES_8_BREAKING` can be used by `@babel/types` consumers to opt-in into stricter validation that we will have in Babel 8. However, opting-in into this AST change makes it impossible to use `@babel/types` with `@babel/parser` and `@babel/generator`.